File: Codes/011_maxArea/11.盛最多水的容器.go
/*
 * @lc app=leetcode.cn id=11 lang=golang
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
func maxArea(height []int) int {

	left := 0
	right := len(height) - 1
	maxValue := -1

	for {
		if  {
			break
		}

		value_left := height[left]
		value_right := height[right]

		var value int
		if value_left > value_right {
			value = (right - left) * value_right
			for ; height[right] <= value_right && right > left; right-- {
			}
		} else {
			value = (right - left) * value_left
			for ; height[left] <= value_left && right > left; left++ {
			}
		}

		if value > maxValue {
			maxValue = value
		}
	}

	return maxValue

}

// @lc code=end

File: Codes/011_maxArea/11.盛最多水的容器.py
#
# @lc app=leetcode.cn id=11 lang=python3
#
# [11] 盛最多水的容器
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left,right=0,len(height)-1

        maxValue=-1

        while left<right:
            h_left,h_right=height[left],height[right]
            if h_left< h_right:
                value=(right-left)*h_left
                while h_left>=height[left] and left<len(height):
                    left+=1
            else:
                value=(right-left)*h_right
                while h_right>=height[right] and right>=0:
                    right-=1
            if value>maxValue: maxValue=value

        return maxValue
                
# @lc code=end

